### PR TITLE
Jax devicearray sum

### LINKED ISSUE #19418: sum

### PR DESCRIPTION
**Pull request by VictorOdede**
Close #19418

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py [hunk 1]
```diff
@@ -593,6 +593,46 @@ def test_jax_devicearray_sort(
     )
 
 
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="jax.numpy.array",
+    method_name="sum",
+    dtype_and_x=helpers.dtype_values_axis(
+        available_dtypes=["int64"],
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py:601]: The tests should also cover all supported dtypes

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @VictorOdede 

The changes are good, some requested changes are below

**Comment by VictorOdede** [comment]
Hey @hirwa-nshuti , I've added tests to include all numeric dtypes

File: ivy/functional/frontends/jax/devicearray.py [hunk 2]
```diff
@@ -138,6 +138,41 @@ def sort(self, axis=-1, order=None):
             order=order,
         )
 
+    def sum(
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/jax/devicearray.py:141]: The promotion here will be done with the same approach as the one mentioned on the `prod` PR.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey

Some other changes needed😊

**Comment by VictorOdede** [comment]
Hey @hirwa-nshuti , the most recent changes to `main` are causing all Jax tests to fail with `AttributeError: module 'jax.numpy' has no attribute 'DeviceArray'`. Not sure what's causing this problem

**Comment by hirwa-nshuti** [comment]
> Hey @hirwa-nshuti , the most recent changes to `main` are causing all Jax tests to fail with `AttributeError: module 'jax.numpy' has no attribute 'DeviceArray'`. Not sure what's causing this problem

Working on a fix for this issue😊

**Pull request closed without merging**